Packs/AggregatedScripts/Scripts/IsolateEndpoint/IsolateEndpoint.py

```diff
@@ -0,0 +1,451 @@
+        self.hard_coded_args = hard_coded_args
+
+
+def initialize_commands() -> list:
+    """
+    Initializes the commands for the Isolate Endpoint.
+    """
+    commands = [
+        Command(
+            # Can be used only in XSIAM
+            brand=Brands.CORTEX_CORE_IR,
+            name="core-isolate-endpoint",
+            arg_mapping={"endpoint_id": "endpoint_id"},
+        ),
+        Command(
+            brand=Brands.CROWDSTRIKE_FALCON,
+            name="cs-falcon-contain-host",
+            arg_mapping={"ids": "endpoint_id"},
+        ),
+        Command(
+            brand=Brands.FIREEYE_HX_V2,
+            name="fireeye-hx-host-containment",
+            arg_mapping={"agentId": "endpoint_id", "hostName": "endpoint_hostname"},  # command can use agentId or hostName
+        ),
+        Command(
+            brand="Microsoft Defender ATP",  # this name is used in get-endpoint-data script,
+            name="microsoft-atp-isolate-machine",
+            arg_mapping={"machine_id": "endpoint_id"},
+            hard_coded_args={"isolation_type": "Full", "comment": "Isolated endpoint with IsolateEndpoint script."},
+        ),
+    ]
+    return commands
+
+
+""" HELPER FUNCTIONS """
+
+
+def check_inputs_for_command(command: Command, endpoint_output: dict, args: dict) -> bool:
+    """
+    Validates whether a command can be executed by checking the required arguments.
+
+    Args:
+        command (Command): An instance containing the command's metadata and argument mapping.
+        endpoint_output (dict): The dictionary to store output results.
+        args (dict): The dictionary containing the specific arguments for the command.
+
+    Returns:
+        bool: True if the command can be executed, False otherwise.
+    """
+    missing_args = are_there_missing_args(command, args)  # checks if there are missing args
+    if missing_args:
+        demisto.debug(f"Missing the next args {missing_args} for command.name")
+        create_message_to_context_and_hr(
+            is_isolated=False,
+            endpoint_args=args,
+            result="Fail",
+            message=f"Missing the next args: {missing_args} for {command.name}.",
+            endpoint_output=endpoint_output,
+        )
+        return False
+    return True
+
+
+def is_endpoint_already_isolated(endpoint_data: dict, endpoint_args: dict, endpoint_output: dict) -> bool:
+    """
+    Checks whether an endpoint is isolated already.
+
+    Args:
+        endpoint_data (dict): A dictionary containing endpoint details.
+        endpoint_args (dict): The arguments used in the command execution.
+        endpoint_output (dict): A list to store structured output results.
+
+    Returns:
+        bool: True if the endpoint is isolated, False otherwise.
+    """
+    demisto.debug(f"Got endpoint {endpoint_data} with field isIsolated{endpoint_data.get('IsIsolated')}")
+    is_isolated = endpoint_data.get("IsIsolated", "No")
+    if is_isolated == "Yes":
```

Comment:
You can do here instead:
    if is_isolated != "Yes":
        return False
So to not go into the nesting logic if not necessary. 
and then the rest, with return True at the end